code/modules/mob/living/simple_animal/hostile/retaliate/pet.dm

```diff
@@ -0,0 +1,23 @@
+/mob/living/simple_animal/hostile/retaliate/araneus
```

Comment:
And then it won't retaliate because they're in the same faction, and AFAIK a mob will never attack someone that's in their own faction even when provoked.

On top of that, that still leaves the issue that the giant_spider file is 270 lines of code full of abilities this this pet doesn't need. The amount of work to get rid of all these special abilities and make the /giant_spider/ subtype actually act like a pet would probably be as much as the solution I'm using.